### PR TITLE
Remove unwrap() and fix clippy warnings

### PR DESCRIPTION
**Pull request by JasonTulp**
# Description

- This PR makes changes to the entire codebase based on running cargo clippy. Changes have been made where deemed appropriate. It also removes the use of unwrap() in favour of expect (with the exception of in mock/ test code) 

## Context & Background

- The ticket to remove unwrap was brought up in TRN-211 from Red4Sec which is what this PR aims to address

## Related Issues

- https://futureverse.atlassian.net/browse/TRN-230?atlOrigin=eyJpIjoiNjkzZWM3YjU1ZmRjNDIzNGI3MWY0NmM5ZDVjMDEwMjQiLCJwIjoiaiJ9

---

# Release Notes

## Type of Change

- [x] Runtime Changes
- [ ] Client Changes

### CODE REVIEW

**Comment by zees-dev** [comment]
CI fix required

**Comment by zees-dev** [comment]
As part of this PR the CI should also be updated to enforce cargo clippy compliancy for the codebase.

File: ethy-gadget/rpc/src/lib.rs [hunk 1]
```diff
@@ -132,8 +132,8 @@ where
 		if let Ok(maybe_encoded_proof) = self.client.get_aux(
 			[
 				ETHY_ENGINE_ID.as_slice(),
-				&[EthyChainId::Ethereum.into()].as_slice(),
-				&event_id.to_be_bytes().as_slice(),
+				([EthyChainId::Ethereum.into()].as_slice()),
+				(event_id.to_be_bytes().as_slice()),
```
Comment (zees-dev) [inline ethy-gadget/rpc/src/lib.rs:136]: unsure how this is still compiling as its made a tuple 😅

File: evm-precompiles/dex/src/lib.rs [hunk 2]
```diff
@@ -99,47 +99,44 @@ where
...
-				Action::Quote => Self::quote(handle),
-				Action::GetAmountIn => Self::get_amount_in(handle),
-				Action::GetAmountOut => Self::get_amount_out(handle),
-				Action::GetAmountsIn => Self::get_amounts_in(handle),
-				Action::GetAmountsOut => Self::get_amounts_out(handle),
-			}
+		let selector = match handle.read_selector() {
+			Ok(selector) => selector,
+			Err(e) => return Err(e.into()),
 		};
```
Comment (zees-dev) [inline evm-precompiles/dex/src/lib.rs:105]: nit: probably could use `let Ok(selector) = handle.read_selector() else {` syntax

File: pallet/nft/src/impls.rs [hunk 3]
```diff
@@ -377,6 +373,7 @@ impl<T: Config> Pallet<T> {
 		(new_cursor, total_owned, response)
 	}
 
+	//noinspection ALL
```
Comment (zees-dev) [inline pallet/nft/src/impls.rs:376]: unsure why this is needed; seeing it in a few places

File: pallet/xrpl-bridge/src/lib.rs [hunk 4]
```diff
@@ -1520,36 +1516,37 @@ impl<T: Config> Pallet<T> {
...
-				tx_fee,
-				SourceTag::<T>::get(),
-				// omit signer key since this is a 'MultiSigner' tx
-				None,
-			);
-			payment.binary_serialize(true)
-		};
-
-		tx_blob
+		match destination_tag {
```
Comment (zees-dev) [inline pallet/xrpl-bridge/src/lib.rs:1519]: can use `if let Some(..) = destination_tag else {` syntax here to reduce nesting.

File: runtime/src/impls.rs [hunk 5]
```diff
@@ -1032,10 +1032,10 @@ impl seed_pallet_common::ExtrinsicChecker for DoughnutCallValidator {
 				keep_alive,
 			}) => {
 				let asset_id_u64: u64 = (*asset_id).into();
-				let who = <Runtime as frame_system::Config>::Lookup::lookup(destination.clone())
+				let who = <Runtime as frame_system::Config>::Lookup::lookup(*destination)
 					.map_err(|_| pallet_doughnut::Error::<Runtime>::ToppingPermissionDenied)?;
 				let destination: [u8; 20] = who.into();
-				let amount_u128: u128 = (*amount).into();
+				let amount_u128: u128 = *amount;
```
Comment (zees-dev) [inline runtime/src/impls.rs:1038]: redundant var imo

**Review by zees-dev — COMMENTED** [review]
Great stuff; lots of code improvements here! 🚀

File: evm-precompiles/dex/src/lib.rs (hunk 2 above)
Comment (JasonTulp) [reply evm-precompiles/dex/src/lib.rs:105]: This doesn't work as you need the error from within the Result

File: ethy-gadget/rpc/src/lib.rs (hunk 1 above)
Comment (JasonTulp) [reply ethy-gadget/rpc/src/lib.rs:136]: I think it just added brackets, either way probably unnecessary, will remove

File: pallet/ethy/src/impls.rs [hunk 6]
```diff
@@ -326,13 +325,14 @@ impl<T: Config> Pallet<T> {
...
-			log!(error, "💎 eth_getTransactionReceipt({:?}) failed: {:?}", tx_hash, err);
-			return EventClaimResult::DataProviderErr;
-		}
+		let maybe_tx_receipt = match T::EthereumRpcClient::get_transaction_receipt(tx_hash) {
+			Ok(r) => r,
+			Err(err) => {
+				log!(error, "💎 eth_getTransactionReceipt({:?}) failed: {:?}", tx_hash, err);
+				return EventClaimResult::DataProviderErr;
+			},
+		};
```
[the remark this answers is not in the document]
Comment (JasonTulp) [reply pallet/ethy/src/impls.rs:334]: Won't work here either, need the error from the result

File: pallet/nft/src/impls.rs (hunk 3 above)
Comment (JasonTulp) [reply pallet/nft/src/impls.rs:376]: If we wanted to automate Clippy, we'd need this all through the codebase for items we don't want to get picked up

Comment (JasonTulp) [reply pallet/nft/src/impls.rs:376]: But personally I think this is a bad idea, will remove these comments for now

File: runtime/src/impls.rs (hunk 5 above)
Comment (JasonTulp) [reply runtime/src/impls.rs:1038]: Haha agreed

File: pallet/xrpl-bridge/src/lib.rs (hunk 4 above)
Comment (JasonTulp) [reply pallet/xrpl-bridge/src/lib.rs]: https://github.com/futureversecom/trn-seed/pull/902/commits/753c1ac1f8e87f860e3911e540cb4f8d94971798

File: pallet/nft/src/impls.rs (hunk 3 above)
Comment (JasonTulp) [reply pallet/nft/src/impls.rs]: https://github.com/futureversecom/trn-seed/pull/902/commits/753c1ac1f8e87f860e3911e540cb4f8d94971798

File: pallet/ethy/src/impls.rs [hunk 7]
```diff
@@ -326,13 +325,14 @@ impl<T: Config> Pallet<T> {
...
 
-		let maybe_tx_receipt = result.unwrap(); // error handled above qed.
 		let tx_receipt = match maybe_tx_receipt {
```
[the remark this answers is not in the document]
Comment (JasonTulp) [reply pallet/ethy/src/impls.rs]: https://github.com/futureversecom/trn-seed/pull/902/commits/753c1ac1f8e87f860e3911e540cb4f8d94971798

File: evm-precompiles/erc721/src/lib.rs [hunk 8]
```diff
@@ -974,9 +965,7 @@ where
 					.write(Address::from(Into::<H160>::into(collection_info.owner)))
 					.build(),
 			)),
-			None => {
-				Err(revert(alloc::format!("ERC721: Collection does not exist").as_bytes().to_vec()))
-			},
+			None => Err(revert(String::from("ERC721: Collection does not exist").as_bytes())),
```
[the remark this answers is not in the document]
Comment (JasonTulp) [reply evm-precompiles/erc721/src/lib.rs]: https://github.com/futureversecom/trn-seed/pull/902/commits/753c1ac1f8e87f860e3911e540cb4f8d94971798

File: evm-precompiles/erc721/src/lib.rs [hunk 9]
```diff
@@ -725,9 +718,7 @@ where
 					.write::<U256>(collection_info.collection_issuance.into())
 					.build(),
 			)),
-			None => {
-				Err(revert(alloc::format!("ERC721: Collection does not exist").as_bytes().to_vec()))
-			},
+			None => Err(revert(String::from("ERC721: Collection does not exist").as_bytes())),
```
[the remark this answers is not in the document]
Comment (JasonTulp) [reply evm-precompiles/erc721/src/lib.rs]: https://github.com/futureversecom/trn-seed/pull/902/commits/753c1ac1f8e87f860e3911e540cb4f8d94971798

File: evm-precompiles/erc721/src/lib.rs [hunk 10]
```diff
@@ -659,9 +656,7 @@ where
 					.write::<Bytes>(collection_info.name.as_slice().into())
 					.build(),
 			)),
-			None => {
-				Err(revert(alloc::format!("ERC721: Collection does not exist").as_bytes().to_vec()))
-			},
+			None => Err(revert(String::from("ERC721: Collection does not exist").as_bytes())),
```
[the remark this answers is not in the document]
Comment (JasonTulp) [reply evm-precompiles/erc721/src/lib.rs]: https://github.com/futureversecom/trn-seed/pull/902/commits/753c1ac1f8e87f860e3911e540cb4f8d94971798

File: evm-precompiles/erc721/src/lib.rs [hunk 11]
```diff
@@ -300,7 +301,7 @@ where
 					.write(Address::from(Into::<H160>::into(owner_account_id)))
 					.build(),
 			)),
-			None => Err(revert(alloc::format!("ERC721: Token does not exist").as_bytes().to_vec())),
+			None => Err(revert(String::from("ERC721: Token does not exist").as_bytes())),
```
[the remark this answers is not in the document]
Comment (JasonTulp) [reply evm-precompiles/erc721/src/lib.rs]: https://github.com/futureversecom/trn-seed/pull/902/commits/753c1ac1f8e87f860e3911e540cb4f8d94971798

**Review by zees-dev — APPROVED** [review]
LGTM 👍

File: pallet/assets-ext/src/weights.rs [hunk 12]
```diff
@@ -62,14 +62,14 @@ impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
 	// Storage: Futurepass DefaultProxy (r:1 w:0)
 	// Storage: Assets Metadata (r:1 w:1)
 	fn create_asset() -> Weight {
-		Weight::from_all(175_374_000 as u64)
-			.saturating_add(T::DbWeight::get().reads(7 as u64))
-			.saturating_add(T::DbWeight::get().writes(6 as u64))
+		Weight::from_all(175_374_000_u64)
```
Comment (surangap) [inline pallet/assets-ext/src/weights.rs:65]: benchmarking might say otherwise😅

Comment (surangap) [reply pallet/assets-ext/src/weights.rs:65]: alright found this -> https://github.com/futureversecom/trn-seed/pull/902/files#diff-1735c4f20d5fa959838642b22b9c1f4eeaf7acae6d5de9508f6914b106dc7b98R70

**Review by surangap — APPROVED** [review]
lgtm. was fun reviewing this. Thanks for working on it.

**Pull request merged**